presto-thrift-connector/src/main/java/com/facebook/presto/connector/thrift/ThriftSessionProperties.java

```diff
@@ -27,15 +28,26 @@
 public final class ThriftSessionProperties
 {
     private final List<PropertyMetadata<?>> sessionProperties;
+    private static final String SET_THRIFT_IDENTITY_HEADER = "THRIFT_CONNECTOR_SET_IDENTITY_HEADERS";
 
     @Inject
     public ThriftSessionProperties(ThriftConnectorConfig config)
     {
-        sessionProperties = ImmutableList.of();
+        PropertyMetadata<Boolean> s1 = PropertyMetadata.booleanProperty(
+                SET_THRIFT_IDENTITY_HEADER,
+                "Set to true to set thrift identity. Default false.",
+                config.shouldSetIdentityThriftHeaders(),
+                false);
+        this.sessionProperties = ImmutableList.of(s1);
```

Comment:
I would inline this, like so (note the static import of `booleanProperty`, as is convention elsewhere, e.g. `HiveSessionPropertes):

```suggestion
        sessionProperties = ImmutableList.of(
            booleanProperty(
                    SET_THRIFT_IDENTITY_HEADER,
                    "Set to true to set thrift identity. Default false.",
                    config.shouldSetIdentityThriftHeaders(),
                    false));
```